src/main/java/echoquery/sql/SchemaInferrer.java

```diff
@@ -102,20 +105,44 @@ private void adjustColumnToTable(String table, DatabaseMetaData md)
 
   /**
    *
+   * @param columns the columns of interest
+   * @param sourceTable the table of interest
+   * @param destinationTable the table being joined
+   * @return an ordered list of which table each column belongs to. the ith
+   *         table in prefixes corresponds to the ith column in columns.
+   */
+  private String[] inferPrefixes(String[] columns, String sourceTable, String destinationTable) {
+    String[] prefixes = new String[columns.length];
+    for (int i = 0; i < columns.length; i++) {
+      String column = columns[i];
+      Set<String> tables = columnToTable.get(column);
```

Comment:
we need to be careful here if there are many columns of the same name but different table - maybe thats an assumption that's made by star schema but seems like could be an easy one to mess up in real data